Gems/ROS2/Code/Tests/GNSSTest.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzTest/AzTest.h>
+
+#include "GNSS/GNSSFormatConversions.h"
+
+namespace UnitTest
+{
+
+    class GNSSTest : public AllocatorsTestFixture
+    {
+    };
+
+    TEST_F(GNSSTest, WGS84ToECEF)
+    {
+        const AZStd::vector<AZStd::pair<AZ::Vector3, AZ::Vector3>> input_gold_set = {
```

Comment:
The variable name should use lowerCamelCase.
```suggestion
        const AZStd::vector<AZStd::pair<AZ::Vector3, AZ::Vector3>> inputGoldSet = {
```